src/main/java/com/icthh/xm/uaa/service/TenantRoleService.java

```diff
@@ -142,7 +145,21 @@ public void addRole(@Valid RoleDTO roleDto) {
             enrichExistingPermissions(permissions, roleDto.getRoleKey(), roleDto.getBasedOn());
         }
 
-        updatePermissions(mapper.writeValueAsString(permissions));
+        updatePermissions(permissions);
+    }
+
+    private void removeDefaultValues(Map<String, Map<String, Set<Permission>>> permissions) {
+        if (TRUE.equals(tenantPropertiesService.getTenantProps().getSecurity().getRemoveDefaultPermissions())) {
+            permissions.forEach((serviceName, service) ->
```

Comment:
Seems it could be refactored in a pure functional way with no inner lambdas: 
```
permissions.values()
                       .stream()
                       .map(Map::values)
                       .flatMap(Collection::stream)
                       .forEach(role -> role.removeIf(shouldRemovePermission()));
```

and extract Predicate to the method:
```
private Predicate<Permission> shouldRemovePermission(){
        return permission -> permission.isDisabled()
                     && permission.getEnvCondition() == null
                     && permission.getResourceCondition() == null
                     && (permission.getReactionStrategy() == null || permission.getReactionStrategy() == EXCEPTION);
    }
```